eZ/Publish/Core/Repository/RoleService.php

```diff
@@ -121,8 +126,386 @@ public function __construct(
     }
 
     /**
+     * Creates a new RoleDraft.
+     *
+     * @since 6.0
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to create a role
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the name of the role already exists or if limitation of the
+     *                                                                        same type is repeated in the policy create struct or if
+     *                                                                        limitation is not allowed on module/function
+     * @throws \eZ\Publish\API\Repository\Exceptions\LimitationValidationException if a policy limitation in the $roleCreateStruct is not valid
+     *
+     * @param \eZ\Publish\API\Repository\Values\User\RoleCreateStruct $roleCreateStruct
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\RoleDraft
+     */
+    public function createRoleDraft(APIRoleCreateStruct $roleCreateStruct)
+    {
+        if (!is_string($roleCreateStruct->identifier) || empty($roleCreateStruct->identifier)) {
+            throw new InvalidArgumentValue('identifier', $roleCreateStruct->identifier, 'RoleCreateStruct');
+        }
+
+        if ($this->repository->hasAccess('role', 'create') !== true) {
+            throw new UnauthorizedException('role', 'create');
+        }
+
+        try {
+            $existingRoleDraft = $this->loadRoleDraftByIdentifier($roleCreateStruct->identifier);
```

Comment:
Seems to only check role drafts, while it also needs to check published roles, so maybe change the spi here to reflect that.